### PR TITLE
Rendering within distance optimisation

### PR DESCRIPTION
**Pull request by PrzemekTomczyk**

### CODE REVIEW

File: ARGO/src/InputSystem.cpp [hunk 1]
```diff
@@ -17,10 +17,11 @@ void InputSystem::update(Entity& t_entity)
 	//std::cout << "Calling InputComponent.update()" << std::endl;
 #endif
 
-	if (t_entity.getAllComps().at(COMPONENT_ID::INPUT_ID) && t_entity.getAllComps().at(COMPONENT_ID::COMMAND_ID))
+	InputComponent* inputComp = static_cast<InputComponent*>(t_entity.getComponent(ComponentType::Input));
+	CommandComponent* commandComp = static_cast<CommandComponent*>(t_entity.getComponent(ComponentType::Command));
+
+	if (inputComp != nullptr && commandComp != nullptr)
```
Comment (Violet-Byrne) [inline ARGO/src/InputSystem.cpp:23]: would (!inputComp  && !commandComp) not be better

File: ARGO/src/RenderSystem.cpp [hunk 2]
```diff
@@ -161,6 +170,21 @@ void RenderSystem::renderText(SDL_Renderer* t_renderer, TransformComponent* t_po
...
 }
 
+bool RenderSystem::inView(TransformComponent* t_posComp)
+{
+	float left, right, top, down;
+	left = m_focusPoint.x - Utilities::SCREEN_WIDTH * 0.75f;
+	right = m_focusPoint.x + Utilities::SCREEN_WIDTH * 0.6f;
+	top = m_focusPoint.y - Utilities::SCREEN_HEIGHT * 0.75f;
+	down = m_focusPoint.y + Utilities::SCREEN_HEIGHT * 0.6f;
+	if (left < t_posComp->getPos().x && right > t_posComp->getPos().x&&
```
Comment (CathalRedmond) [inline ARGO/src/RenderSystem.cpp]: return this

**Review by CathalRedmond — CHANGES_REQUESTED** [review]

File: ARGO/src/RenderSystem.cpp [hunk 3]
```diff
@@ -18,29 +18,38 @@ void RenderSystem::render(SDL_Renderer* t_renderer, Entity& t_entity)
...
+			TextComponent* textComp = static_cast<TextComponent*>(t_entity.getComponent(ComponentType::Text));
+
+			if (visComp != nullptr)
+			{
+				renderTexture(visComp, posComp->getPos().x, posComp->getPos().y, t_renderer);
+			}
+			if (textComp != nullptr)
+			{
+				renderText(t_renderer, posComp, textComp);
+			}
```
Comment (Violet-Byrne) [inline ARGO/src/RenderSystem.cpp:33]: (visComp) or (textComp) not be better?

File: ARGO/src/RenderSystem.cpp [hunk 4]
```diff
@@ -161,6 +170,21 @@ void RenderSystem::renderText(SDL_Renderer* t_renderer, TransformComponent* t_po
 	SDL_SetRenderDrawColor(t_renderer, prevRGBA[0], prevRGBA[1], prevRGBA[2], prevRGBA[3]);
 }
 
+bool RenderSystem::inView(TransformComponent* t_posComp)
+{
+	float left, right, top, down;
+	left = m_focusPoint.x - Utilities::SCREEN_WIDTH * 0.75f;
+	right = m_focusPoint.x + Utilities::SCREEN_WIDTH * 0.6f;
+	top = m_focusPoint.y - Utilities::SCREEN_HEIGHT * 0.75f;
+	down = m_focusPoint.y + Utilities::SCREEN_HEIGHT * 0.6f;
```
Comment (Violet-Byrne) [inline ARGO/src/RenderSystem.cpp]: Why is left .75 of the screens width and right .6
also could hold this value as a variable in the header

**Review by Emmett-Byrne — CHANGES_REQUESTED** [review]

**Review by Emmett-Byrne — APPROVED** [review]

**Pull request merged**